File: frontend/src/types/events.ts
export interface Event {
  id: number;
  title: string;
  description?: string; // Made optional since it's not in your API response
  startAt: string;
  endAt: string;
  registrationOpensAt: string;
  registrationClosesAt: string;
  eventType: string; // Changed from eventTypeId to eventType to match API
  eventTypeId: number;
  status: 'REGISTERED' | 'NOT_REGISTERED' | 'CANCELED'; // Changed from registrationStatus to status
  guestCount: number;
  createdBy?: number; // Made optional since it's not in your API response
  createdAt?: string; // Made optional since it's not in your API response
  isActive?: boolean; // Made optional since it's not in your API response
}

export interface EventsResponse {
  data: Event[]; // Changed from events to data to match API response
  pagination: {
    totalItem: number;
    totalPage: number;
    currentPage: number;
  }; // Changed from totalPage to pagination object
}

export interface RegisterEventRequest {
  action: 'REGISTER' | 'CANCEL' | 'NOT_REGISTERED';
  guestCount: number;
}

export interface EventFilters {
  mode?: 'ALL' | 'ONGOING' | 'UPCOMING' | 'RECENT';
  page?: number;
  limit?: number;
  sortBy?: string;
  sortOrder?: 'ASC' | 'DESC';
} 

File: backend/internal/entity/events.go
package entity

import "time"

type Events struct {
	Id                    int        `db:"id"                          json:"id"`
	Title                 string     `db:"title"                       json:"title"`
	Description           string     `db:"description"                 json:"description"`
	EventTypeId           int        `db:"event_type_id"               json:"eventTypeId"`
	StartAt               time.Time  `db:"start_at"                    json:"startAt"`
	RegistrationOpensAt   time.Time  `db:"registration_opens_at"       json:"registrationOpensAt"`
	RegistrationClosesAt  time.Time  `db:"registration_closes_at"      json:"registrationClosesAt"`
	ShouldAutoCreateEvent bool       `db:"should_auto_create_event"    json:"shouldAutoCreateEvent"`
	TotalParticipants     int        `db:"total_participants"          json:"totalParticipants"`
	MaxParticipants       int        `db:"max_participants"            json:"maxParticipants"`
	IsActive              bool       `db:"is_active"                   json:"isActive"`
	IsDeleted             bool       `db:"is_deleted"                  json:"isDeleted"`
	CreatedBy             *int       `db:"created_by"                  json:"createdBy"`
	UpdatedBy             *int       `db:"updated_by"                  json:"updatedBy"`
	Remarks               *string    `db:"remarks"                     json:"remarks"`
	CreatedAt             time.Time  `db:"created_at"                  json:"createdAt"`
	UpdatedAt             *time.Time `db:"updated_at"                  json:"updatedAt"`
}
